server/lib/imageproxy.ts

```diff
@@ -187,16 +211,25 @@ class ImageProxy {
       const directory = join(this.getCacheDirectory(), cacheKey);
       const href =
         this.baseUrl +
-        (this.baseUrl.endsWith('/') ? '' : '/') +
+        (this.baseUrl.length > 0
+          ? this.baseUrl.endsWith('/')
+            ? ''
+            : '/'
+          : '') +
         (path.startsWith('/') ? path.slice(1) : path);
       const response = await this.fetch(href);
       const arrayBuffer = await response.arrayBuffer();
       const buffer = Buffer.from(arrayBuffer);
 
-      const extension = path.split('.').pop() ?? '';
-      const maxAge = Number(
+      const extension = (response.headers.get('content-type') ?? '').replace(
```

Comment:
What is the purpose of modifying the extension to content-type?
Won't this create issues with the `writeToCacheDir` function that uses it for the filename?

Reply:
I had to do it this way because the URL of the profile pictures from Jellyfin and Emby don't have a file extension, causing issues. However, I recognize that getting the image file extension from the content-type header needs to be handled better, because by just using .replace('/image', '') can cause further issues.